src/github/github.installation.service.ts

```diff
@@ -0,0 +1,173 @@
+import { Injectable, Logger, NotFoundException } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { AllowedOrganization } from './entities/allowed-organization.entity';
+import { Repository } from 'typeorm';
+import { OrganizationStatus } from 'src/common/constants';
+import { CreateOrgDto } from './dto/create.org.dto';
+import { Env } from 'src/env';
+import * as fs from 'fs';
+import * as path from 'path';
+import * as jwt from 'jsonwebtoken';
+import { Octokit } from '@octokit/rest';
+
+@Injectable()
+export class GithubInstallationService {
+  private readonly logger = new Logger(GithubInstallationService.name);
+  constructor(
+    @InjectRepository(AllowedOrganization)
+    private allowedOrgRepository: Repository<AllowedOrganization>,
+  ) {}
+
+  async create(createOrgDto: CreateOrgDto): Promise<AllowedOrganization> {
+    const newOrg = this.allowedOrgRepository.create(createOrgDto);
+    return this.allowedOrgRepository.save(newOrg);
+  }
+
+  async findAll(): Promise<AllowedOrganization[]> {
+    return this.allowedOrgRepository.find();
+  }
+
+  async findOne(id: string): Promise<AllowedOrganization> {
+    const org = await this.allowedOrgRepository.findOne({ where: { id } });
+    if (!org) {
+      throw new NotFoundException(`Organization not found`);
+    }
+    return org;
+  }
+
+  async findByName(organizationName: string): Promise<AllowedOrganization> {
+    const org = await this.allowedOrgRepository.findOne({
+      where: { organizationName },
+    });
+    if (!org) {
+      throw new NotFoundException(
+        `Organization "${organizationName}" not found`,
+      );
+    }
```

Comment:
PullProofAI

Suggest consolidating error handling logic for finding an organization by ID and name to avoid repetition.
```diff
+     async findOneOrFail(id: string): Promise<AllowedOrganization> {
+         const org = await this.allowedOrgRepository.findOne({ where: { id } });
+         if (!org) {
+             throw new NotFoundException(`Organization not found`);
+         }
+         return org;
+     }
+
+     async findByNameOrFail(organizationName: string): Promise<AllowedOrganization> {
+         const org = await this.allowedOrgRepository.findOne({ where: { organizationName } });
+         if (!org) {
+             throw new NotFoundException(`Organization "${organizationName}" not found`);
+         }
+         return org;
+     }
```



<!-- This is an auto-generated comment by PullProofAI -->